Add RNTupleTempSource and RNTupleTempOutputModule

Dr15Jones opened the pull request:
#### PR description:

- Used PoolSource and PoolOutputModule as templates for their equivalent implementation using RNTuple. The new modules are meant to be used to further stress test the use of RNTuple in CMSSW. A different set of modules will be made for use during HL-LHC running.
- moved commits from RNTUPLE_X build to master.

#### PR validation:

Code compiles. Unit tests that pass in IBs pass after change.

resolves https://github.com/cms-sw/framework-team/issues/1680

Dr15Jones committed:
Moved RunHelper to FWCore/Sources and renamed

This will allow other sources to reuse the code

Dr15Jones committed:
All RNTupleTemp related files

Names have changed but still using TTree for now.

Dr15Jones committed:
Replaced meta data TTrees with RNTuple

- Also removed some backwards compatibility code

Dr15Jones committed:
Added FWIO/RNTupleTempTests items
  - based on IOPool/Tests
  - includes changes needed to make tests work

Dr15Jones committed:
Removed cloning code from RNTuple implementation

Cloning is not an option for RNTuple.

Dr15Jones committed:
Moved output to use only RNTuple

Dr15Jones committed:
Simplifying input code for RNTuple migration

Dr15Jones committed:
edmRNTupleTempFileUtil now complete

Dr15Jones committed:
Updated edmRNTupleTempProvDump to only use RNTuple

Removed all references to TTree used during the transition.

Dr15Jones committed:
Removed unneeded header file

Dr15Jones committed:
Move input to only use RNTuple

Dr15Jones committed:
Better configuration of RNTupleTemp*

- made TTree only parameters optional
- added RNTuple specific options

Dr15Jones committed:
Update to use std::format

Dr15Jones committed:
Handle RNTuple optimizations

- properly pass optimizations to RNTupleWriter
- can specify per Field optimizations

Dr15Jones committed:
Use config to decide meta data compression in RNTupleTemp

Dr15Jones committed:
Remove redundant FixMissingStreamerInfos

Dr15Jones committed:
Update TestRNTupleTempGetBy to new Tracer output

The Tracer service added new state transition output so the results
of the test need to be updated.

Dr15Jones committed:
Added missing header

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones for master.

It involves the following packages:

- Configuration/PyReleaseValidation (**pdmv**)
- FWCore/Sources (**core**)
- FWIO/RNTuple (**core**)
- FWIO/RNTupleTempInput (**core**)
- FWIO/RNTupleTempOutput (**core**)
- FWIO/RNTupleTempTests (**core**)
- IOPool/Common (**core**)
- IOPool/Input (**core**)
- PhysicsTools/NanoAOD (**xpog**)


@AdrianoDee, @DickyChant, @Dr15Jones, @antoniovagnerini, @battibass, @ftorrresd, @makortel, @miquork, @smuzaffar can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @fabiocos, @gpetruc, @makortel, @missirol, @mmusich, @slomeo, @wddgit this is something you requested to watch as well.
@ftenchini, @mandrenguyen, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
-1

**Failed Tests**:   ClangBuild 
**Size**: This PR adds an extra 80KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-56d9da/49771/summary.html
**COMMIT**: 5839e3b3639da8018b20654fa98b69a6de4a1ad8
**CMSSW**: CMSSW_16_0_X_2025-12-02-1100/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/49532/49771/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Failed Clang Build

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' /usr/bin/time -v scram build -k -j 32 COMPILER='llvm compile'

```
 See details on the summary page.

cmsbuild commented:
Pull request has been put on hold by  @Dr15Jones
They need to issue an `unhold` command to remove the `hold` state or L1 can `unhold` it for all

Dr15Jones commented:
unhold

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49532/47043

- Found files with invalid states:
  - FWIO/RNTupleTempInput/test/testNoParentDictionary.sh:
    - Added: fd9ee5f1d49531d5f815550288603dbf57f2a957
    - Deleted: 4ca6c132777e3d239359835fbf4c4438664819e6
  - FWIO/RNTupleTempInput/src/FixMissingStreamerInfos.cc:
    - Added: fd9ee5f1d49531d5f815550288603dbf57f2a957
    - Deleted: 8fa87a86844842beb4e9f27abebdb387ea5df091

cmsbuild commented:
Pull request #49532 was updated. @Dr15Jones, @makortel, @smuzaffar can you please check and sign again.

cmsbuild commented:
-1

**Failed Tests**:   ClangBuild 
**Size**: This PR adds an extra 48KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-56d9da/49773/summary.html
**COMMIT**: f673be0f48effe68070ba3fbb1a4b6fd728282c8
**CMSSW**: CMSSW_16_0_X_2025-12-02-1100/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/49532/49773/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Failed Clang Build

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' /usr/bin/time -v scram build -k -j 32 COMPILER='llvm compile'

```
 See details on the summary page.

Dr15Jones committed:
Fix clang compilation issue

The default assignment/copy methods do not work because of
restrictions from the member data.

Dr15Jones commented:
please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Size**: This PR adds an extra 52KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-56d9da/49774/summary.html
**COMMIT**: b3bf9f0f60b38c9e705b27a97d4c3ea95f8a1927
**CMSSW**: CMSSW_16_0_X_2025-12-02-1100/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/49532/49774/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Failed Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test TestRNTupleTempEventHistory had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially [removed 4 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-12-02-1100+56d9da/72125/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 12 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-12-02-1100+56d9da/72125/validateJR.html)
 - Reco comparison had  4 failed jobs
 - DQMHistoTests: Total files compared: 53
 - DQMHistoTests: Total histograms compared: 4269233
 - DQMHistoTests: [Total failures: 73](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-12-02-1100+56d9da/72125/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4269140
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 52 files compared)
 - Checked 227 log files, 198 edm output root files, 53 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-12-02-1100+56d9da/72125/triggerResults): no differences found

makortel commented:
Comparison differences are related to https://github.com/cms-sw/cmssw/issues/47071

makortel commented:
It is a bit weird though that the test with https://github.com/cms-sw/cmssw/pull/49531 still showed a failure in `TestRNTupleTempEventHistory`. If the failure persists in IBs, I think it would be better to investigate there.
